src/query.rs

```diff
@@ -470,6 +483,40 @@ impl WMIConnection {
 
         self.raw_query(&query)
     }
+
+    // Execute the given query in async way, returns result in a Sink.
+    #[cfg(feature = "async-query")]
+    fn exec_async_query_native_wrapper(
+        &self,
+        query: impl AsRef<str>,
+    ) -> Result<ComPtr<IWbemObjectSink>, WMIError> {
+        let query_language = BStr::from_str("WQL")?;
+        let query = BStr::from_str(query.as_ref())?;
+
+        unsafe {
+            // ExecQueryAsync needs a *mut ptr, so we encapsulate in a Box to keep a ref to the sink
+            let pp_sink = Box::new(QuerySink::new_ptr());
+
+            if let Err(e) = check_hres((*self.svc()).ExecQueryAsync(
+                query_language.as_bstr(),
+                query.as_bstr(),
+                WBEM_FLAG_BIDIRECTIONAL as i32,
+                ptr::null_mut(),
+                *pp_sink,
+            )) {
+                // We need to manually release the Sink in case of error
+                (**pp_sink).Release();
```

Comment:
Is this needed?
Reading https://docs.rs/com-impl/0.1.1/com_impl/struct.Refcount.html,
I would guess that auto-decr will handle this?